log4j-core/src/test/java/org/apache/logging/log4j/core/config/LoggerContextChangeTest.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.logging.log4j.core.config;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.Set;
+
+public class LoggerContextChangeTest {
+    private static String targetFileName = "log4j2-3182.xml";
+    private static String targetErrorFileName = "log4j2-3182-error.xml";
+    private static String tmpFileName = "log4j2-3182-tmp.xml";
+    private static String destDir = "target/test-classes/";
+
+    @BeforeAll
+    public static void beforeClass() {
+        System.setProperty("log4j2.configurationFile", "classpath:" + targetFileName);
+    }
+
+
+    @Test
+    public void onChangeTest() {
+        String errorFileName = destDir + targetErrorFileName;
+        String originFileName = destDir + targetFileName;
+        String tmpFile = destDir + tmpFileName;
+        wait(10);
+        updateConfigFileModTime(originFileName, errorFileName, tmpFile);
+        wait(30);
```

Comment:
Given that the current test suite takes more than 30 minutes, I am reluctant to add a test that waits for 10+30=40 seconds.

Reply:
OK. I will reduce this wait time